.github/workflows/v1.yaml

```diff
@@ -21,16 +21,21 @@ jobs:
         working-directory: '1.10'
     steps:
       - uses: actions/checkout@v4
+      - name: Determine composer version from Dockerfile
+        run: |
+          echo "COMPOSER_VERSION_MAJOR=$(grep -oP 'COMPOSER_VERSION \d+\.\d+\.\d+' Dockerfile | sed '/COMPOSER_VERSION /s///' | grep -oP '^\d+')" >> $GITHUB_ENV
+          echo "COMPOSER_VERSION_MAJOR_MINOR=$(grep -oP 'COMPOSER_VERSION \d+\.\d+\.\d+' Dockerfile | sed '/COMPOSER_VERSION /s///' | grep -oP '^\d+\.\d+')" >> $GITHUB_ENV
+          echo "COMPOSER_VERSION_MAJOR_MINOR_PATCH=$(grep -oP 'COMPOSER_VERSION \d+\.\d+\.\d+' Dockerfile | sed '/COMPOSER_VERSION /s///')" >> $GITHUB_ENV
```

Comment:
Exposing the environment variables here

- allows for reuse
- increases the transparency of what values are used where